packages/block-editor/src/components/block-list/block.js

```diff
@@ -231,7 +233,8 @@ function BlockListBlock( {
 				'is-content-block': hasContentLockedParent && isContentBlock,
 			},
 			dataAlign && themeSupportsLayout && `align${ dataAlign }`,
-			className
+			className,
+			layoutClassNames
```

Comment:
We need to add these here for blocks that don't use `InnerBlocks` to render their children (the only examples I could find of this in Core are Gallery and PostContent when viewed in non-editable mode in the site editor, but there may be more). Unfortunately that means that if the block's inner wrapper is different from its outer, the layout classnames will be duplicated across both wrappers in the editor. There don't seem to be any consequences to this, as the blocks that currently have multiple wrappers and layout (e.g. Navigation) are already  handling the styles being applied to their outer wrapper. But it would be nice to fix if anyone has ideas!

Reply:
Because the constrained layout and root padding features have rules that target the direct child of the layout classname, I think we might need to come up with a way where the layout classes are only applied to the deepest node (that is, the innerBlocks wrapper, or the outer wrapper if no inner wrapper is in use). I'm not quite sure how we ensure that, though! 😅

Reply:
The only alternative I can see here is to be deliberate about layout _not_ supporting out of the box blocks that don't use InnerBlocks, and add some custom logic to make things work in Gallery and Post Content. I'll look into that!